### PR TITLE
fix: [DEL-4436]: Replaced simpleTagInputs component with MultiSelect component

### PR DESCRIPTION
**Pull request by amitsh14**
#### Summary

The DelegateSelectors component uses the SimpleTagInput component to show the delegate-selectors values in the dropdown. SimpleTagInput only supports an array of strings and with the latest changes in the delegate selectors API, the requirement of supporting an array of objects was not getting fulfilled. This PR replaces DelegateSelectors' SimpleTagInput with the MutliSelect and delegate selectors API

#### Changes

https://user-images.githubusercontent.com/101694909/185836728-d2271e64-91bc-4dcf-b458-e14c19b79f8c.mov

#### Screenshots

**Current screen:**
Screenshot 2022-08-12 at 2 12 25 PM

**After Change:**
Screenshot 2022-08-22 at 9 27 09 AM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 33 Issues 
* Code issue 33 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10571)

File: src/modules/10-common/components/DelegateSelectors/DelegateSelectors.types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+export interface DelegateSelector {
+  name: string
+  connected: boolean
```
Comment (deepesh-ui) [inline src/modules/10-common/components/DelegateSelectors/DelegateSelectors.types.ts:10]: can we rename to isConnected

File: src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2.tsx [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  placeholder?: string
+  pollingInterval?: number
+  wrapperClassName?: string
+  onTagInputChange?: (tags: string[]) => void
+  data: DelegateSelector[]
+}
+
+const mapSelectedData = (delegateSelectors: DelegateSelector[], selectedItems: string[]) => {
+  const items = delegateSelectors.filter((item: DelegateSelector) => {
+    return selectedItems?.includes(item?.name || '')
```
Comment (deepesh-ui) [inline src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2.tsx:49]:
```suggestion
    return selectedItems?.includes(defaultTo(item?.name,''))
```

File: src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2Container.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    orgIdentifier,
+    projectIdentifier,
+    pollingInterval = null,
+    onTagInputChange = noop,
+    wrapperClassName,
+    placeholder,
+    selectedItems
+  } = props
+
+  // console.log('received:', selectedItems)
```
Comment (deepesh-ui) [inline src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2Container.tsx:41]: remove commented code

Comment (amitsh14) [reply src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2Container.tsx]: done

File: src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2.tsx (hunk 2 above)
Comment (amitsh14) [reply src/modules/10-common/components/DelegateSelectors/DelegateSelectorsV2.tsx]: done

File: src/modules/10-common/components/DelegateSelectors/DelegateSelectors.types.ts (hunk 1 above)
Comment (amitsh14) [reply src/modules/10-common/components/DelegateSelectors/DelegateSelectors.types.ts]: removed types file

**Comment by amitsh14** [comment]
retrigger jest

**Comment by amitsh14** [comment]
retrigger coverage

**Comment by amitsh14** [comment]
retrigger jest

**Comment by amitsh14** [comment]
retrigger coverage

**Comment by amitsh14** [comment]
retrigger coverage

**Comment by amitsh14** [comment]
retrigger coverage

**Review by dhruvUpadhyay07 — APPROVED** [review]

**Review by vkbansal-harness — APPROVED** [review]

**Pull request merged**